tapir/coop/views/shareowner.py

```diff
@@ -283,6 +288,38 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class ShareOwnerDeleteView(
+    LoginRequiredMixin,
+    PermissionRequiredMixin,
+    generic.DeleteView,
+):
+    permission_required = PERMISSION_GROUP_MANAGE
+    model = ShareOwner
+
+    def get_object(self, queryset=None):
+        return super().get_object(queryset)
+
+    def get_success_url(self):
+        return reverse("coop:shareowner_list")
+
+    def dispatch(self, request, *args, **kwargs):
+        obj = self.get_object()
+        if request.user == obj.user:
+            return HttpResponseForbidden("You cannot delete your own account.")
+        return super().dispatch(request, *args, **kwargs)
+
+    @transaction.atomic
+    def form_valid(self, form):
+        share_owner = self.get_object()
+        self.get_object().delete()
```

Comment:
If we keep the override of `delete` as soft_delete then we don't need to call `delete()` manually here: DeleteView does it already in `super().form_valid(form)`.

That's why you don't get the same error I had: I was testing before your changes, where `delete` was the hard delete. So `DeleteView.form_valid` would hard delete the object.